transport-classes-io_uring/src/main/java/io/netty/channel/uring/IOUringDatagramChannel.java

```diff
@@ -0,0 +1,667 @@
+            return block(
+                    multicastAddress,
+                    NetworkInterface.getByInetAddress(localAddress().getAddress()),
+                    sourceToBlock, promise);
+        } catch (Throwable e) {
+            promise.setFailure(e);
+        }
+        return promise;
+    }
+
+    @Override
+    protected AbstractUnsafe newUnsafe() {
+        return new IOUringDatagramChannelUnsafe();
+    }
+
+    @Override
+    protected void doBind(SocketAddress localAddress) throws Exception {
+        if (localAddress instanceof InetSocketAddress) {
+            InetSocketAddress socketAddress = (InetSocketAddress) localAddress;
+            if (socketAddress.getAddress().isAnyLocalAddress() &&
+                    socketAddress.getAddress() instanceof Inet4Address) {
+                if (socket.family() == SocketProtocolFamily.INET6) {
+                    localAddress = new InetSocketAddress(LinuxSocket.INET6_ANY, socketAddress.getPort());
+                }
+            }
+        }
+        super.doBind(localAddress);
+        active = true;
+    }
+
+    private static void checkUnresolved(AddressedEnvelope<?, ?> envelope) {
+        if (envelope.recipient() instanceof InetSocketAddress
+                && (((InetSocketAddress) envelope.recipient()).isUnresolved())) {
+            throw new UnresolvedAddressException();
+        }
+    }
+
+    @Override
+    protected Object filterOutboundMessage(Object msg) {
+        if (msg instanceof DatagramPacket) {
+            DatagramPacket packet = (DatagramPacket) msg;
+            checkUnresolved(packet);
+            ByteBuf content = packet.content();
+            return !content.hasMemoryAddress() ?
+                    packet.replace(newDirectBuffer(packet, content)) : msg;
+        }
+
+        if (msg instanceof ByteBuf) {
+            ByteBuf buf = (ByteBuf) msg;
+            return !buf.hasMemoryAddress()? newDirectBuffer(buf) : buf;
+        }
+
+        if (msg instanceof AddressedEnvelope) {
+            @SuppressWarnings("unchecked")
+            AddressedEnvelope<Object, SocketAddress> e = (AddressedEnvelope<Object, SocketAddress>) msg;
+            checkUnresolved(e);
+            if (e.content() instanceof ByteBuf &&
+                (e.recipient() == null || e.recipient() instanceof InetSocketAddress)) {
+
+                ByteBuf content = (ByteBuf) e.content();
+                return !content.hasMemoryAddress()?
+                        new DefaultAddressedEnvelope<>(
+                            newDirectBuffer(e, content), (InetSocketAddress) e.recipient()) : e;
+            }
+        }
+
+        throw new UnsupportedOperationException(
+                "unsupported message type: " + StringUtil.simpleClassName(msg) + EXPECTED_TYPES);
+    }
+
+    @Override
+    public DatagramChannelConfig config() {
+        return config;
+    }
+
+    @Override
+    protected void doDisconnect() throws Exception {
+        // TODO: use io_uring for this too...
```

Comment:
as ditto before: better to have an issue, unless closed in this pr

Reply:
again... same as we have in the incubator version. We can create an issue once this is in.